### PR TITLE
Increase speed computing picking_count in point of sale sessions

### PR DESCRIPTION
**Pull request by mbaena**
Description of the issue/feature this PR addresses:
Increase speed computing picking_count in point of sale sessions.

Current behavior before PR:
point of sale sessions is loading slowly for a lot pickings with state != 'done'.

Desired behavior after PR is merged:
Increase the speed in loading.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by gdgellatly** [comment]
You could probably do this a lot faster converting the not in to an in of
the desired states.

File: addons/point_of_sale/models/pos_session.py [hunk 1]
```diff
@@ -101,9 +101,14 @@ def _confirm_orders(self):
 
     @api.multi
     def _compute_picking_count(self):
+        ids = set(pos.config_id.picking_type_id.id for pos in self)
+        data = self.env['stock.picking'].read_group(
```
[the remark this answers is not in the document]
Comment (mbaena) [reply addons/point_of_sale/models/pos_session.py]: Hi, I use `set` and  `list` to reduce the number of values. Does `read_group` make this reduction?

[the remark this answers is not in the document]
Comment (fmdl) [reply addons/point_of_sale/models/pos_session.py]: the mapped function return a recorset with dictinct record (with no duplicate).

[the remark this answers is not in the document]
Comment (mbaena) [reply addons/point_of_sale/models/pos_session.py]: the proposed code doesn't work. revert to original PR

**Comment by fmdl** [comment]
hi @mbaena the good code for the conversion is :
in this case i use session_id to count because you have 1 picking for one session.

```
    @api.multi
    def _compute_picking_count(self):
        data = self.env['pos.order'].read_group(
            [('picking_id.state', '!=', 'done'), ('session_id', 'in', self.ids)],
            ['session_id'], ['session_id'])
        count = dict([(x['session_id'][0], x['session_id_count']) for x in data])
        for pos in self:
            pos.picking_count = count.get(pos.id, 0)
```

**Pull request closed without merging**